File: 4week/남규/3. Two City Scheduling.cpp
// https://leetcode.com/problems/two-city-scheduling/

class Solution {
public:
    int twoCitySchedCost(vector<vector<int>>& costs) {
        priority_queue<vector<int>> pq;
        for(auto cost : costs){
            pq.push({cost[0] - cost[1], cost[0], cost[1]});
        }
        
        int n = costs.size() / 2;
        int ans = 0;
        
        for(int i = 0; i < n; i++){
            ans += pq.top()[2];
            pq.pop();
        }
        
        for(int i = 0; i < n; i++){
            ans += pq.top()[1];
            pq.pop();
        }
        return ans;
    }
};

File: 4week/홍준/3. two-city-scheduling.py
class Solution:
    def twoCitySchedCost(self, costs: List[List[int]]) -> int:
        answer = 0
        n = len(costs) // 2
        sorted_costs = sorted(costs, key=lambda x: x[0]-x[1])
        for i in range(n):
            answer += sorted_costs[i][0]
        for i in range(n, 2*n):
            answer += sorted_costs[i][1]
        return answer
        